lib/logic/auth/child/sign_up/child_sign_up_state.dart

```diff
@@ -10,18 +10,18 @@ class ChildSignUpState extends FormzState {
 	ChildSignUpState({
 		this.caregiverCode = const UserCode.pure(),
 		this.name = const Name.pure(),
-		this.avatar = 0,
+		this.avatar,
 		this.takenAvatars = const {},
 		FormzStatus status = FormzStatus.pure
 	}) : super(status);
 
 	@override
 	List<Object> get props => [caregiverCode, name, avatar];
 
-	ChildSignUpState copyWith({UserCode caregiverCode, Name name, int avatar, Set<int> takenAvatars, FormzStatus status}) {
+	ChildSignUpState copyWith({UserCode caregiverCode, Name name, int avatar, Set<int> takenAvatars, FormzStatus status, bool clearableAvatar = false}) {
 		return ChildSignUpState(
 			caregiverCode: caregiverCode ?? this.caregiverCode,
-			avatar: avatar ?? this.avatar,
+			avatar: clearableAvatar ? avatar : (avatar ?? this.avatar),
```

Comment:
Why do we need this clearableAvatar param?

Reply:
When changing the caregiver id, potential new set of _takenAvatars_ is set. If user picked avatar first (and then caregiver id with some _takenAvatars_) I need to reset the field. The problem is, with previous _copyWith_ method I can't do that (operator ?? cancels the new value). That's why there is additional parameter for clearing the avatar (setting it to null).

Reply:
Makes sense